rbd: remove unneeded updateSnapshotDetails() function

nixpanic opened the pull request:
updateSnapshotDetails() calls getImageInfo() on the rbdVolume that is
related to the rbdSnapshot. This is not needed at all, as both types
have the same getImageInfo() function, so it can be called directly on
the rbdSnapshot.

nixpanic committed:
rbd: remove unneeded updateSnapshotDetails() function

updateSnapshotDetails() calls getImageInfo() on the rbdVolume that is
related to the rbdSnapshot. This is not needed at all, as both types
have the same getImageInfo() function, so it can be called directly on
the rbdSnapshot.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
@mergifyio rebase

nixpanic commented:
@mergifyio queue

nixpanic commented:
**mini-e2e-helm_k8s-1.29** failed with:
```
[FAILED] backend images not matching kubernetes resource count,image count 6 kubernetes resource count 3
  backend image Info:
   [csi-snap-0cb3a66f-12fb-4de3-abc8-358c9652b925 csi-snap-253ee5c4-5629-4aee-ad41-5b73d46473b9 csi-snap-c8339722-e817-4a42-8fdb-b7d7e484c596 csi-vol-13284667-a769-4ece-9702-da14bccad413 csi-vol-29544488-b299-4b66-bda7-59cfa5c339ea csi-vol-cd3f6a6b-7d6f-4339-8abf-aaa653a1fdc4]
```

[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.29/detail/mini-e2e-helm_k8s-1.29/151/pipeline)

And **mini-e2e-helm_k8s-1.27** as well:
```
[FAILED] backend images not matching kubernetes resource count,image count 6 kubernetes resource count 3
  backend image Info:
   [csi-snap-00c89343-7d46-47c4-92a9-a126bcd81781 csi-snap-c4cba576-fc9b-4603-846b-f2523b6bc7dd csi-snap-c4fbb8cd-37b9-459e-98ef-50d02c2eae90 csi-vol-aebc4d25-5b16-4cd7-9f44-c4b5c117c316 csi-vol-d2399942-4d6e-4585-af59-7136d18acc9e csi-vol-d8a3ce9f-f614-467c-815f-2d59d9d3dcbc]
```

[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.27/detail/mini-e2e-helm_k8s-1.27/802/pipeline)

Madhu-1 commented on internal/rbd/rbd_util.go:
```diff
@@ -1042,33 +1042,14 @@ func genSnapFromSnapID(
-// updateSnapshotDetails will copy the details from the rbdVolume to the
-// rbdSnapshot. example copying size from rbdVolume to rbdSnapshot.
-func updateSnapshotDetails(rbdSnap *rbdSnapshot) error {
-	vol := rbdSnap.toVolume()
```
this call is missing above?

nixpanic replied:
It really should not be needed, because `rbdSnapshot` is an `rbdImage` that implements the `getImageInfo()` function...

nixpanic replied:
The problem is with `getImageInfo()`, it should be called on the parent image, without the `@snapshot` name:
```
failed to update snapshot details for "replicapool/csi-vol-b3a9c24a-6aea-41f1-a221-6fee32da150e@csi-snap-c8339722-e817-4a42-8fdb-b7d7e484c596": image not found: RBD image not found
```

nixpanic committed:
rbd: add rbdSnapshot.open() as alternative to rbdImage.open()

When opening an rbdSnapshot errors are reported that the image can not
be found. Opening the RBD-image with the RbdSnapName things should work.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.27

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.28

nixpanic commented:
> /test ci/centos/mini-e2e/k8s-1.27

This one is still running, but there are `an operation with the given Volume ID 0001-0024-72956532-8149-4e58-8636-8f8d656a3c8d-0000000000000002-294bdff5-caa2-468c-86d0-82974fa340e4 already exists` in the [logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.27/detail/mini-e2e_k8s-1.27/838/pipeline/). Unclear if that is something that is introduced with the new `rbdSnapshot.open()` function.

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.28

nixpanic commented:
Not continuing with this. `rbdSnapshot.RbdSnapName` is too much (ab)used to with relation to the existing `RbdImageName`, there does not seem to be a rule when `RbdSnapName` or when `RbdImageName` is supposed to be used. Only using `RbdSnapName` for the rbd-snapshot and `RbdImageName` for the rbd-image seems too tricky to do.
